docs/book/src/plugins/external-plugins.md

```diff
@@ -0,0 +1,155 @@
+# Extending Kubebuilder with external plugins
+
+## Overview
+
+You can extend the Kubebuilder functionality by introducing external plugin.
+
+An external plugin is an executable that provides the same kubebuilder plugin features. Since it is an executable, such customized plugin can be implemented in any language.
+
+The Kubebuilder CLI loads the external plugin in the specified path and interact the plugin through the executable's `stdin` & `stdout`.
+
+## When is it useful?
+
+- If you want to create helpers or addons on top of the scaffolds done by the Kubebuilder internal plugins.
```

Comment:
Should we still use the terminology as "internal plugins" since all plugins will be implemented as external plugins?

Reply:
+1 on avoiding "internal plugins" as terminology here. I would probably say something more along the lines of "Kubebuilder's default scaffolding"